server/fixity.go

```diff
@@ -139,3 +155,118 @@ func (s *RESTServer) scanfixity() {
 	}
 	log.Println("Ending scanfixity. duration = ", time.Now().Sub(starttime))
 }
+
+func (s *RESTServer) GetFixityHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	item := r.FormValue("item")
+	start := r.FormValue("start")
+	end := r.FormValue("end")
+	status := r.FormValue("status")
+
+	log.Println("GetFixityHandler called")
+	log.Println("item= ", item)
+	log.Println("start= ", start)
+	log.Println("end= ", end)
+	log.Println("status= ", status)
+
+	startValue, startErr := startValidate(start)
+	if startErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", startErr.Error())
+		return
+	}
+
+	endValue, endErr := endValidate(end)
+	if endErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", endErr.Error())
+		return
+	}
+
+	itemValue, itemErr := itemValidate(item)
+	if itemErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", itemErr.Error())
+		return
+	}
+
+	statusValue, statusErr := statusValidate(status)
+	if statusErr != nil {
+		w.WriteHeader(400)
+		log.Println("GetFixityHandler :", statusErr.Error())
+		return
+	}
+
+	fixityResults := s.Fixity.GetFixity(startValue, endValue, itemValue, statusValue)
+	if fixityResults == nil {
+		w.WriteHeader(404)
```

Comment:
Does `GetFixity` returning `nil` mean there are no results? If so, just return an empty JSON list with status 200. There is nothing erroneous with there being no results for a set of conditions.

Reply:
Done.